crates/db_schema/src/impls/post.rs

```diff
@@ -440,13 +440,30 @@ impl PostActions {
   pub async fn mark_as_unread(
     pool: &mut DbPool<'_>,
     form: &PostReadForm,
+  ) -> LemmyResult<UpleteCount> {
+    Self::mark_many_as_unread(pool, std::slice::from_ref(form)).await
+  }
+
+  pub async fn mark_many_as_unread(
+    pool: &mut DbPool<'_>,
+    forms: &[PostReadForm],
   ) -> LemmyResult<UpleteCount> {
     let conn = &mut get_conn(pool).await?;
 
+    // Grab the first person id
+    let person_id = forms
+      .iter()
+      .map(|f| f.person_id)
+      .next()
+      .ok_or_else(|| LemmyErrorType::CouldntUpdate)?;
```

Comment:
This is going to break if someone tries to mark posts from different users as read for some reason. Better to make this explicit and change the params to `PersonId, Vec<PostId>`.

It would also be nice to merge mark_many_as_unread() and mark_many_as_read() into one, but that seems impossible now because of uplete.

Reply:
> It would also be nice to merge mark_many_as_unread() and mark_many_as_read() into one, but that seems impossible now because of uplete.

If you're talking about marking some posts as read and some posts as unread in the same function, I don't see how that would be useful.

If it's because of the different return types, the solution is calling `UpleteCount::only_updated` when marking posts as read.